### DISCUSSION TITLE
Assert failed:	invalid base for 'eTime.4' [rtl_433_stats:246:31] -- what does that mean?

### DISCUSSION THREAD

**Discussion by hdtodd** [exaloop/codon#287] [Q&A] [answered]
I thought I'd try Codon on a ISM-band analysis program I wrote (https://github.com/hdtodd/rtl_433_stats , in the py-version directory).  

I'm new to Python, and Codon was helpful in pointing out a couple of situations in which I was misusing the language.  But the next error, with the message above, is one I just can't interpret, though I think it's suggesting that I'm trying to act on two variables of mismatched types.

eTime is a float constructed from a datetime string:
`            eTime = datetime.datetime.fromisoformat(y['time']).timestamp()`
firstTime is a float:
`firstTime = float('inf')`
(I've tried 1000000.0 instead of 'inf', too, in case there was some issue with 'inf'.  Same result.)

I've confirmed that during execution in Python (with type()) the two are both floats.  So I don't see what Codon is choking on.

The offending line is:
`        firstTime = eTime if (eTime<firstTime) else firstTime`
I've also tried min() and got the same message.

Does anyone have a clue what I've done wrong?

**Comment by elisbyberi** [comment]
Would you be able to provide a minimal reproducible example that can aid in debugging this error?

**Comment by hdtodd** [comment, marked as answer]
@elisbyberi ,

I can't reproduce it, but I can point to the issue.  I tried to create a minimal example, below.  It failed to compile until I inserted the line `lc = 1` at the start of the code, above the function definition.  Even with the `global lc` in the function definition, if I just defined `lc = 1` at the main program, after the function definition, codon errored-out and issued a message `name 'lc' is not defined` (three times: one for the global definition and two for the use in print statements).

I went back to my original program and defined the offending variables BEFORE any function definitions, and those compile-time errors were resolved.  Codon complained that `min` was used before it was defined, so I replaced the calls to min() and max() with the prior conditional assignment statements.

But Codon then complained that `pktcount` wasn't defined in my class init (`self.pktcount=1`) until I removed `self.`.  Ditto for the other class variables in the `__init__`.  I don't think the resulting code would perform as intended, but I removed all those `self.` prefixes, and now Codon complains that "'str' object has no attribute 'format'".  So it still doesn't compile, and I use string formatting heavily in the print statements.

As a result of  these edits, my code base is now pretty far from the working Python version, and I'm not sure I could reconcile the two back into one code base.  So I'm going to put aside attempts to compile with Codon for now.  

```
#/!usr/bin/env python3
# demo.py
# Demonstrate Codon compile failure

import datetime
lc = 1

##########################################################################################
# Convert time string ("ts") from ISO or epoch to float epoch, with either
#  integer or fractional seconds
def CnvTime(ts):
    global lc
    if ts.find("-") > 0:
        try:
            eTime = datetime.datetime.fromisoformat(ts).timestamp()
        except ValueError as e:
            err={}
            print("datetime error in input line", lc, "converting time string: ", ts)
            print("datetime msg:", err.get("error", str(e)))
            exit(1)
    else:
        try:
            eTime = float(ts)
        except ValueError as e:
            err = {}
            print("Datetime conversion failed on line", lc, "with datetime string", ts)
            print("float() error msg:", err.get("error", str(e)))
            exit(1)
    return eTime

firstTime = float('inf')

eTime = CnvTime("2023-02-26 07:31:32")

print("Before min:", "firstTime=", firstTime, "eTime=", eTime)

firstTime = min(eTime, firstTime)

print("After min:","firstTime=", firstTime, "eTime=", eTime)
```
